keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/support/ExtensionRegistry.kt

```diff
@@ -0,0 +1,24 @@
+package com.netflix.spinnaker.keel.api.support
+
+/**
+ * Registers an extension to a model so that it may be used in a delivery config.
+ */
+interface ExtensionRegistry {
+  fun <BASE : Any> register(
+    baseType: Class<BASE>,
+    extensionType: Class<out BASE>,
+    discriminator: String
+  )
+
+  fun <BASE : Any> extensionsOf(baseType: Class<BASE>): Map<String, Class<out BASE>>
+}
```

Comment:
This interface is part of `keel-api` so plugins can use it to add additional sub-types that will get properly registered with Jackson and springdoc-openapi without requiring the plugin to have either of those as dependencies.